kong/db/strategies/postgres/init.lua

```diff
@@ -1386,7 +1386,7 @@ function _M.new(connector, schema, errors)
       })
 
       local conflict_key = unique_escaped
-      if has_composite_cache_key then
+      if has_composite_cache_key and not unique_field.is_endpoint_key then
```

Comment:
When doing the operation upsert by field, if that field is used as endpoint_key, then it is most likely unique(based Kong RESTful style), and should be used as SQL conflict condition, instead of using the composite cache_key.